src/networking.c

```diff
@@ -840,13 +845,20 @@ void addReplyDouble(client *c, double d) {
                               d > 0 ? 6 : 7);
         }
     } else {
-        char dbuf[MAX_LONG_DOUBLE_CHARS+3],
-             sbuf[MAX_LONG_DOUBLE_CHARS+32];
-        int dlen, slen;
+        char dbuf[MAX_LONG_DOUBLE_CHARS+32];
+        int dlen = 0;
         if (c->resp == 2) {
-            dlen = snprintf(dbuf,sizeof(dbuf),"%.17g",d);
-            slen = snprintf(sbuf,sizeof(sbuf),"$%d\r\n%s\r\n",dlen,dbuf);
-            addReplyProto(c,sbuf,slen);
+            // Preserve chars before the double for $0000\r\n, and trim unused.
```

Comment:
please use block comment, not line comment (one of the few things redis conventions are consistent about).
please elaborate a bit more about what it does and why.
i suppose the why is to avoid copying the possibly huge string twice.
and the what is that you reserve room for a big header, and then place it seasoned to the data and skip the extra space when you send the reply.